Tests/Common/TestUtils.cs

```diff
@@ -182,5 +185,107 @@ public static TimeSpan GetStartTimeOffsetForSecondWindow()
+            {
+                throw new TimeoutException("Killing the process via SSH took more than expected.");
+            }
+        }
+
+        public static void StartBasicsStation(TestConfiguration config, Dictionary<string, string> scriptParameters, out string randomDirectoryName)
+        {
+            if (config is null) throw new ArgumentNullException(nameof(config));
+
+            var connection = config.RemoteConcentratorConnection;
+            var sshPrivateKeyPath = config.SshPrivateKeyPath;
+            var rootPath = TryGetSolutionDirectoryInfo().FullName;
+            var loRaBasicsStationModulePath = Path.Combine(rootPath, "LoRaEngine", "modules", "LoRaBasicsStationModule");
+            var scriptName = "start_basicsstation.sh";
+            var startScript = new FileInfo(Path.Combine(loRaBasicsStationModulePath, scriptName));
+            var helperFunctions = new FileInfo(Path.Combine(loRaBasicsStationModulePath, "helper-functions.sh"));
+            var stationConf = new FileInfo(Path.Combine(loRaBasicsStationModulePath, "station.conf"));
+
+            // Copying needed files in a local temporary path
+            var localTempPath = Path.GetTempPath();
+            randomDirectoryName = Path.GetRandomFileName();
+            var tempDirectory = Path.Combine(localTempPath, randomDirectoryName);
+            _ = Directory.CreateDirectory(tempDirectory);
+            File.Copy(config.BasicStationExecutablePath, Path.Combine(tempDirectory, "station.std"));
+            File.Copy(startScript.FullName, Path.Combine(tempDirectory, scriptName));
+            File.Copy(helperFunctions.FullName, Path.Combine(tempDirectory, "helper-functions.sh"));
+            File.Copy(stationConf.FullName, Path.Combine(tempDirectory, "station.conf"));
+
+            Environment.SetEnvironmentVariable("BS_TEMP_DIRECTORY", tempDirectory);
+            // Following environment variable is needed if following bash commands are executed within WSL
+            Environment.SetEnvironmentVariable("WSLENV", "BS_TEMP_DIRECTORY/up");
+
+            if (!config.RunningInCI)
+            {
+                using var folderCreateProcess = new Process()
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = "bash",
+                        Arguments = $"-c \"scp -i {sshPrivateKeyPath} -rp $BS_TEMP_DIRECTORY {connection}:/tmp\"",
+                        RedirectStandardOutput = true,
+                        UseShellExecute = false,
+                        CreateNoWindow = true,
+                    }
+                };
+                _ = folderCreateProcess.Start();
+                if (!folderCreateProcess.WaitForExit(30_000))
+                {
+                    throw new TimeoutException("Creating and copying over SSH the needed files took more than expected.");
+                }
+                tempDirectory = $"/tmp/{randomDirectoryName}";
```

Comment:
Nit: could we output this tempdirectory variable so if needed we could fetch the basic station log from the ci

Reply:
Temporary directory gets cleaned up while disposing the test class, therefore we can't fetch the logs afterwards.
We could decide to log always everything, but it could be very verbose.

Let's keep this discussion separate from this PR and keep this as an improvement point for the future